src/collection_management.rs

```diff
@@ -133,14 +133,18 @@ pub trait CollectionManagement:
         let caller = self.blockchain().get_caller();
         self.require_is_privileged(&caller);
         if !self.freezed_addresses_for_collection().contains(&address) {
-            let total_freezed = self.freezed_sfts_per_address(address).push(&nonce);
-            self.freezed_count(&address).set(&total_freezed);
-            self.set_freezed_sfts_per_address_event(&address, nonce);
-            self.freeze_single_nft(nonce, &address)
-                .async_call()
-                .call_and_exit();
+            if self.freezed_sfts_per_address(&address).insert(nonce) {
+                let total_freezed = self.freezed_sfts_per_address(&address).len();
+                self.freezed_count(&address).set(&total_freezed);
+                self.set_freezed_sfts_per_address_event(&address, nonce);
+                self.freeze_single_nft(nonce, &address)
+                    .async_call()
+                    .call_and_exit();
+            } else {
+                sc_panic!("Nonce not found in freeze list");
```

Comment:
@bucurdavid is this accurate?

Why would it hit this panic line? 
`sc_panic!("Nonce not found in freeze list");`

Reply:
It should be this message:
`sc_panic!("Nonce already in the freeze list");`